Firestore/core/src/firebase/firestore/api/document_snapshot.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_API_DOCUMENT_SNAPSHOT_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_API_DOCUMENT_SNAPSHOT_H_
+
+#if !defined(__OBJC__)
+#error "This header only supports Objective-C++"
+#endif  // !defined(__OBJC__)
+
+#import <Foundation/Foundation.h>
+
+#include <string>
+#include <utility>
+
+#import "Firestore/Source/Model/FSTFieldValue.h"
+
+#include "Firestore/core/src/firebase/firestore/model/document_key.h"
+#include "Firestore/core/src/firebase/firestore/model/field_path.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class FIRFirestore;
+@class FIRSnapshotMetadata;
+@class FIRDocumentReference;
+@class FSTDocument;
+
+namespace firebase {
+namespace firestore {
+namespace api {
+
+class DocumentSnapshot {
```

Comment:
High-level comment: you seem to be trying to hide the implementation of this class, but I think it's unnecessary. This isn't a public interface.

In any case where the body of a method can be simply specified inline, wouldn't it be easier to just do so?